### PR TITLE
neocna: add support for Azure Server and FlexibleServer (MySQL)

### PR DESCRIPTION
**Pull request by kcarrasco-panw**
Resolves [CNS-5910](https://redlock.atlassian.net/browse/CNS-5910) by adding the needed attributes to support Azure MySQL Flexible Server and Server (single instance).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,15 +64,13 @@ lint:
...
+		--exclude=package-comments \
 		--enable=errcheck \
 		--enable=goimports \
 		--enable=ineffassign \
 		--enable=revive \
 		--enable=unused \
-		--enable=structcheck \
 		--enable=staticcheck \
-		--enable=varcheck \
-		--enable=deadcode \
```
Comment (t00f) [inline Makefile:75]: What is the reason behind the removal of these linters?

File: specs/azureresource.spec [hunk 2]
```diff
@@ -82,6 +84,7 @@ attributes:
     - MicrosoftCompute
     - MicrosoftNetwork
     - MicrosoftDocumentDB
+    - MicrosoftDBforMySQL
```
Comment (t00f) [inline specs/azureresource.spec:87]: Would `MicrosoftMySQL` make sense?

**Review by t00f — COMMENTED** [review]
Thanks for your contribution! 🙏  
I've added a few comments/questions.
Also, can you add a small description so we have the context while looking at the file diffs please?

File: Makefile (hunk 1 above)
Comment (kcarrasco-panw) [reply Makefile:75]: The removed linters are deprecated and covered by the existing `unused` linter.

The exclusion of the `package-comments` linter fixes the CI checks with Go 1.18 and 1.19. Those versions complain about some missing package comments (not modified by this PR). This is new behaviour in the latest release on golangci-lint.

File: specs/azureresource.spec (hunk 2 above)
Comment (kcarrasco-panw) [reply specs/azureresource.spec:87]: I'm following @malkhamis-panw 's lead here and using the name of the provider in the resource ID. eg:
```
/subscriptions/66ca9d9b-c184-4f77-b95d-06f3a5e619b9/resourceGroups/neocna9/providers/Microsoft.DBforMySQL/flexibleServers/mysql-neocna-1
```

File: Makefile (hunk 1 above)
Comment (ericrpowers) [reply Makefile:75]: Yeah, we have to do this to remove annoying golangci-lint warning from v1.49.0+. These changes look good as they are the same I did in a3s repo recently. We can split it from this PR to bring master in sync sooner, or wait for this to merge.

Comment (kcarrasco-panw) [reply Makefile]: I've rebased to master which now includes these changes.

**Comment by kcarrasco-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "neocna-azure-mysql",
    "component": "gaia",
    "pr-id": "838",
    "commit-sha": "ac17c9e60a1e261f3a360a33ebc9f78ea926ab56",
    "pipeline": "master"
  },
  {
    "project": "neocna-azure-mysql",
    "component": "backend",
    "pr-id": "1894",
    "commit-sha": "89ac9ff9a8dea5993afc25b02addbe2e4ad7e645",
    "pipeline": "master"
  }
]
```

**Comment by kcarrasco-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "neocna-azure-mysql",
    "component": "backend",
    "pr-id": "1894",
    "commit-sha": "4e367e3483598f6e52f2e4bf7072c38b9837dd4c",
    "pipeline": "master"
  },
  {
    "project": "neocna-azure-mysql",
    "component": "gaia",
    "pr-id": "838",
    "commit-sha": "ac17c9e60a1e261f3a360a33ebc9f78ea926ab56",
    "pipeline": "master"
  }
]
```

**Comment by kcarrasco-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "neocna-azure-mysql",
    "component": "backend",
    "pr-id": "1894",
    "commit-sha": "5ca32c9a1756d99f717389e0cba77f3b587c45bc",
    "pipeline": "master"
  },
  {
    "project": "neocna-azure-mysql",
    "component": "gaia",
    "pr-id": "838",
    "commit-sha": "ac17c9e60a1e261f3a360a33ebc9f78ea926ab56",
    "pipeline": "master"
  }
]
```

**Comment by kcarrasco-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "neocna-azure-mysql",
    "component": "gaia",
    "pr-id": "838",
    "commit-sha": "ac17c9e60a1e261f3a360a33ebc9f78ea926ab56",
    "pipeline": "master"
  },
  {
    "project": "neocna-azure-mysql",
    "component": "backend",
    "pr-id": "1894",
    "commit-sha": "559eaf078ee85895cdcf6c4e3b5f2785fd85d191",
    "pipeline": "master"
  }
]
```

**Review by malkhamis-panw — APPROVED** [review]
Please,, go ahead and merge; we need this before the backend MR..

Thank you Chris for the initial review

**Pull request merged**